torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -678,6 +678,141 @@ bool ProcessGroupNCCL::CoalescedWorkNCCL::wait(
 
 static std::atomic<size_t> process_group_id = 0;
 
+ProcessGroupNCCL::ProcessGroupNCCL(
+    DistributedBackendOptions& opts,
+    c10::intrusive_ptr<Options> options)
+    : Backend(opts.group_rank, opts.group_size),
+      store_(opts.store),
+      options_(options),
+      ncclCommCounter_(0),
+      traceKeyStart_(getTraceStartKey("NCCL", opts.group_rank)),
+      traceKeyEnd_(getTraceEndKey("NCCL", opts.group_rank)),
+      terminateProcessGroup_(false),
+      terminateHeartbeatMonitorThread_(false),
+      collectiveDebugInfoMode_(false),
+      uid_(process_group_id++) {
+  TORCH_CHECK_WITH(
```

Comment:
Can we consolidate the logic with the existing constructor?

Reply:
That's what I mean, so if we can make the old point to this new constructor, that will also be great.